### PR TITLE
IGAPP-109: Use URL to get Pathnames

### PR DESCRIPTION
**Pull request by Taggotty**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: native/src/modules/app/createNavigateToInternalLink.js [hunk 1]
```diff
@@ -22,24 +23,25 @@ export const createNavigateToInternalLink = ({
...
 }) => ({ url, language }: NavigateToInternalLinkParamsType) => {
-  const parts = url.split('/').filter(segment => segment)
-  const pathnameParts = parts.splice(2)
+  const parsedUrl = new URL(url)
+  const pathname = parsedUrl.pathname
+  const pathnameParts = pathname.split('/').filter(Boolean)
   const newCity = pathnameParts[0]
   const newLanguage = pathnameParts[1]
-  const pathname = pathnameParts.reduce((acc, part) => `${acc}/${part}`, '')
+  const normalizedPathname = pathnameParts.reduce((acc, part) => `${acc}/${part}`, '')
```
Comment (Taggotty) [inline native/src/modules/app/createNavigateToInternalLink.js:31]: Added this as double slashes are actually allowed in URL to current standard. See [here](https://url.spec.whatwg.org/#dom-url-pathname). Alternatively I tried using `react-native-url-polyfill` and I tried using  `ẁhatwg-url`, but as they both also follow the URL standard directly the result did not change. Do we really need a handling for double slashes in URLs?

I do not feel that this PR is a real improvement. Other opinions?

Comment (klinzo) [reply native/src/modules/app/createNavigateToInternalLink.js:31]: I am quite sure that this was not to normalize double slashes, but rather to just have a pathname at all. So this line is not necessary imo, just use the pathname from above. Previously we did not have a pathname at all which is why we had to create it manually by splitting and splicing from the url.
The only improvement should be that we don't have to split the url manually to remove the `https://integreat.app` to get the pathname which is admittedly really a minor improvement.

Comment (klinzo) [reply native/src/modules/app/createNavigateToInternalLink.js:31]: The filter above is probably not needed as well.

**Review by klinzo — COMMENTED** [review]
Is this behavior tested? Could you add some tests if not please?

File: native/src/modules/app/createNavigateToInternalLink.js (hunk 1 above)
Comment (Taggotty) [reply native/src/modules/app/createNavigateToInternalLink.js]: The filter is needed, as every pathname starts with a `/`. `/augsburg.split('/')` would therefore result in `['', 'augsburg']`

**Review by klinzo — APPROVED** [review]

File: native/src/modules/app/createNavigateToInternalLink.js [hunk 2]
```diff
@@ -22,11 +23,11 @@ export const createNavigateToInternalLink = ({
   navigateToCategory: NavigateToCategoryParamsType => void,
   navigateToDashboard: NavigateToCategoryParamsType => void
 }) => ({ url, language }: NavigateToInternalLinkParamsType) => {
-  const parts = url.split('/').filter(segment => segment)
-  const pathnameParts = parts.splice(2)
+  const parsedUrl = new URL(url)
+  const pathname = parsedUrl.pathname
+  const pathnameParts = pathname.split('/').filter(Boolean)
```
Comment (maxammann) [inline native/src/modules/app/createNavigateToInternalLink.js:28]: This filter filters empty parts, right?

**Review by maxammann — APPROVED** [review]

**Pull request merged**